app/src/main/java/dev/msfjarvis/aps/data/password/PasswordEntry.kt

```diff
@@ -59,23 +71,60 @@ class PasswordEntry(content: String, private val totpFinder: TotpFinder = UriTot
         return Otp.calculateCode(totpSecret, Date().time / (1000 * totpPeriod), totpAlgorithm, digits).get()
     }
 
-    val extraContentWithoutAuthData by lazy(LazyThreadSafetyMode.NONE) {
+    private fun generateExtraContentWithoutAuthData(): String {
         var foundUsername = false
-        extraContent.splitToSequence("\n").filter { line ->
-            return@filter when {
-                USERNAME_FIELDS.any { prefix -> line.startsWith(prefix, ignoreCase = true) } && !foundUsername -> {
-                    foundUsername = true
-                    false
-                }
-                line.startsWith("otpauth://", ignoreCase = true) ||
-                    line.startsWith("totp:", ignoreCase = true) -> {
-                    false
+        return extraContent
+            .lineSequence()
+            .filter { line ->
+                return@filter when {
+                    USERNAME_FIELDS.any { prefix -> line.startsWith(prefix, ignoreCase = true) } && !foundUsername -> {
+                        foundUsername = true
+                        false
+                    }
+                    line.startsWith("otpauth://", ignoreCase = true) ||
+                        line.startsWith("totp:", ignoreCase = true) -> {
+                        false
+                    }
+                    else -> {
+                        true
+                    }
                 }
-                else -> {
+            }.joinToString(separator = "\n")
+    }
+
+    private fun generateExtraContentPairs(): Map<String, String> {
+        fun MutableMap<String, String>.putOrAppend(key: String, value: String) {
+            val existing = this[key]
+            this[key] = if (existing == null) {
+                value
+            } else {
+                "$existing\n$value"
+            }
+        }
+        val items = mutableMapOf<String, String>()
+        extraContentWithoutAuthData
+            .lines()
+            // Split on ':', but pass down the line using a Pair because we may need it later
+            .map { line -> line to line.split(':') }
+            // Only take the item when a pair can be formed
+            .filter { (_, list) -> list.size >= 2 }
+            // Convert them to a Pair so it's easier to perform checks
+            // Since the actual values can also contain colons,
+            // we join them back and trim any initial spaces in them,
+            // and trailing spaces in the keys.
+            .map { (line, list) -> line to (list[0].trimEnd() to list.drop(1).joinToString(":").trimStart()) }
+            // Ensure neither key nor value are empty
+            .filter { (line, pair) ->
+                if (pair.first.isNotBlank() && pair.second.isNotBlank()) {
                     true
+                } else {
+                    items.putOrAppend(EXTRA_CONTENT, line)
+                    false
                 }
             }
```

Comment:
Filter seems kinda counter-intuitive to me here considering the fact that I would assume that if we are returning false then that item is not going into the map but it is actually is going into the map inside the EXTRA_CONTENT pair.

Reply:
false simply means it's not going forward to the next step. Since it needs specialized handling I'd rather handle it this way and keep the terminating statement clean but I'd be fine if you reworked it to send the complexity down to the very bottom.

Reply:
:+1: